src/test/java/co/rsk/bitcoinj/script/P2shErpFederationRedeemScriptParserTest.java

```diff
@@ -327,6 +341,58 @@ public void isP2shErpFed_falseWithCustomRedeemScript() {
         Assert.assertFalse(P2shErpFederationRedeemScriptParser.isP2shErpFed(customRedeemScript.getChunks()));
     }
 
+    @Test
+    public void getMultiSigType_shouldReturnP2shErpFedType() {
+        Assert.assertEquals(MultiSigType.P2SH_ERP_FED, p2shErpFederationRedeemScriptParser.getMultiSigType());
+    }
+
+    @Test
+    public void getM_shouldReturnM() {
+        int expectedM = defaultRedeemScriptKeys.size() / 2 + 1;
+        Assert.assertEquals(expectedM, p2shErpFederationRedeemScriptParser.getM());
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyExists_shouldReturnIndex() {
+        for (int i = 0; i < defaultRedeemScriptKeys.size(); i++) {
+            BtcECKey expectedKey = defaultRedeemScriptKeys.get(i);
+            Assert.assertEquals(i, p2shErpFederationRedeemScriptParser.findKeyInRedeem(expectedKey));
+        }
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyDoesNotExists_shouldThrowIllegalStateException() {
+        BtcECKey unknownKey = BtcECKey.fromPrivate(BigInteger.valueOf(1234567890L));
+        Exception actualException = null;
+        try {
+            p2shErpFederationRedeemScriptParser.findKeyInRedeem(unknownKey);
+        } catch (Exception ex) {
+            actualException = ex;
+        } finally {
+            Assert.assertTrue(actualException instanceof IllegalStateException);
+            Assert.assertTrue(
+                actualException.getMessage().contains("Could not find matching key " + unknownKey
+                    + " in script "));
+        }
+    }
```

Comment:
This is quite strange. What's the idea? Test that it throws an IllegalStateException? You can add that using the annotation `@Test(expected = IllegalStateException.class)`

Reply:
Updated!